File: Sources/APTradeDomain/NewsArticle.swift
import Foundation

/// A single news item, normalized from whatever the news source returns. Pure value type.
public struct NewsArticle: Identifiable, Equatable, Codable, Sendable {
    public let id: String
    public let headline: String
    public let summary: String
    public let source: String
    public let url: URL
    public let imageURL: URL?
    public let publishedAt: Date
    public let category: String?
    public let relatedSymbol: String?

    public init(id: String, headline: String, summary: String, source: String,
                url: URL, imageURL: URL?, publishedAt: Date,
                category: String?, relatedSymbol: String?) {
        self.id = id
        self.headline = headline
        self.summary = summary
        self.source = source
        self.url = url
        self.imageURL = imageURL
        self.publishedAt = publishedAt
        self.category = category
        self.relatedSymbol = relatedSymbol
    }
}

File: shared/src/commonMain/kotlin/com/aptrade/shared/domain/NewsArticle.kt
package com.aptrade.shared.domain

/**
 * A single news item, normalized from whatever the news source returns. Pure value type.
 *
 * Spec divergence from the macOS original: `url`/`imageUrl` are plain strings rather than
 * a `URL` type, and `publishedAtEpochSeconds` is a `Long` rather than a `Date` — invariants
 * (non-blank headline/url, defaulted summary/source, etc.) live in the mapper, not here.
 */
data class NewsArticle(
    val id: String,
    val headline: String,
    val summary: String,
    val source: String,
    val url: String,
    val imageUrl: String?,
    val publishedAtEpochSeconds: Long,
    val category: String?,
    val relatedSymbol: String?,
)
